packages/navigation/src/index.js

```diff
@@ -197,6 +207,48 @@ export function updateQueryString(
 	getHistory().push( newPath );
 }
 
+/**
+ * Adds a listener that runs on history change.
+ *
+ * @param {Function} listener Listener to add on history change.
+ * @return {Function} Function to remove listeners.
+ */
+export const addHistoryListener = ( listener ) => {
```

Comment:
Nice decision to move this over.  I think this may be something that could potentially benefit others.

Reply:
👍🏻